flat/ChildMintableERC721.sol

```diff
@@ -2174,6 +2176,45 @@ contract ChildMintableERC721 is
         _burn(tokenId);
     }
 
+    /**
+     * @notice called when user wants to withdraw token back to root chain with token URI
+     * @dev Should handle withraw by burning user's token.
+     * Should set `withdrawnTokens` mapping to `true` for the tokenId being withdrawn
+     * This transaction will be verified when exiting on root chain
+     *
+     * Before calling this function, you may want calling `encodeTokenMetadata`
+     * and get metadata to be transferred from L2 to L1 during exit
+     *
+     * @param tokenId tokenId to withdraw
+     * @param metaData Metadata which client wants to take along with it, generally return value of `encodeTokenMetadata`
+     */
+    function withdrawWithMetadata(uint256 tokenId, bytes calldata metaData) external {
```

Comment:
I think we shouldn't allow passing whatever data inside this function, it's better to keep such logic within the contract, so the developer has control over data passing back to L1.

Reply:
For example, there are some game-related data within the token, like speed/attack, the advanced user will just set any value he wants, ruining the game.

Reply:
Yes that's correct. Putting metaData encoder function call inside `withdrawWithMetadata` solves our problem. 

What do you think ?

Reply:
Yes, remove it, and enforce metaData encoder

Reply:
> Yes, remove metadata argument, and enforce metaData encoder

Ok, I'll make that change.

Reply:
Made changes.